gpkit/posyarray.py

```diff
@@ -30,33 +28,33 @@ class PosyArray(np.ndarray):
 
     def __str__(self):
         "Returns list-like string, but with str(el) instead of repr(el)."
-        return "["+", ".join([str(p) for p in self])+"]"
+        return "[" + ", ".join(str(p) for p in self) + "]"
 
     def __repr__(self):
         "Returns str(self) tagged with gpkit information."
         return "gpkit.%s(%s)" % (self.__class__.__name__, str(self))
 
     def __hash__(self):
-        if hasattr(self, "_hashvalue"):
-            return self._hashvalue
-        else:
-            return np.ndarray.__hash__(self)
+        return getattr(self, "_hashvalue", np.ndarray.__hash__(self))
 
-    def __new__(cls, input_array, info=None):
+    def __new__(cls, input_array):
         "Constructor. Required for objects inheriting from np.ndarray."
         # Input array is an already formed ndarray instance
-        # We first cast to be our class type
+        # cast to be our class type
         obj = np.asarray(input_array).view(cls)
-        # add the new attribute to the created instance
-        obj.info = info
-        # Finally, we must return the newly created object:
         return obj
 
     def __array_finalize__(self, obj):
         "Finalizer. Required for objects inheriting from np.ndarray."
-        if obj is None:
-            return
-        self.info = getattr(obj, 'info', None)
+        pass
+
+    def __array_wrap__(self, out_arr, context=None):
+        """Called by numpy ufuncs.
+        Special case to avoid creation of 0-dimensional arrays
+        See http://docs.scipy.org/doc/numpy/user/basics.subclassing.html"""
+        if out_arr.ndim:
+            return np.ndarray.__array_wrap__(self, out_arr, context)
+        return float(out_arr)
```

Comment:
Doh - obviously dumb. What's most confusing is *why this worked*.

It seems that numpy catches all Exceptions coming from `__array_wrap__`, which is really sad. In this case, `float(out_arr)` raises an `AttributeError` when `out_arr` is a Signomial. What happens next has me baffled -- numpy catches the error, and then for some odd reason successfully returns a Signomial instead of a 0-dimensional array containing a Signomial.

This happens to be the behavior we want, but is obviously a terrible implementation -- pushed a change, will comment on that diff.